src/routes/stats/StatsView.vue

```diff
@@ -173,21 +173,44 @@ export default {
     },
   },
   created() {
-    io.socket.get('/stats', (res) => {
+    io.socket.get('/stats/seasons/current', (res) => {
       this.loadingData = false;
       if (!res?.length) {
         this.error = true;
         return;
       }
       this.seasons = res;
+
       const seasonId = parseInt(this.$route.params.seasonId);
-      this.checkAndSelectSeason(seasonId);
+      if (seasonId) {
+        this.checkAndSelectSeason(seasonId);
+        return;
+      }
+      [this.selectedSeason] = this.seasons;
     });
   },
   methods: {
     checkAndSelectSeason(seasonId) {
+      this.loadingData = true;
       const requestedSeason = this.seasons.find(({ id }) => id === seasonId);
-      this.selectedSeason = requestedSeason || this.seasons[0];
+
+      if (requestedSeason?.rankings?.length) {
+        this.loadingData = false;
+        this.selectedSeason = requestedSeason;
+        return;
+      }
+
+      io.socket.get(`/stats/seasons/${seasonId}`, ({gameCounts,rankings, uniquePlayersPerWeek}) => {
+        if (!rankings) {
+          this.error = true;
+          this.loadingData = false;
+          return;
+        }
+     
+        this.selectedSeason = { ...requestedSeason, gameCounts, rankings, uniquePlayersPerWeek };
+        this.loadingData = false;
+      });
+      
```

Comment:
I think we could possibly put this in a `try{}` skip the catch and just `finally{this.loadingData = false}` but im unsure if that's the best way